### PR TITLE
docs: updated mutation docs to include instructions and caveats for i…

### LINKED ISSUE #3136: Add documentation for non-root mutations
Namespaced mutations are recommended by apollo and other organizations (TODO: links).

We could provide a small section for how to implement this idea.

See https://github.com/strawberry-graphql/strawberry/issues/2853 for example

### PR DESCRIPTION
**Pull request by bradleyoesch**
docs: updated mutation docs to include instructions and caveats for implementing nested mutations

## Description

Updated mutation docs to include instructions and caveats for implementing nested mutations, as decscribed by [Namespacing by separation of concerns](https://www.apollographql.com/docs/technotes/TN0012-namespacing-by-separation-of-concern/).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* Closes #3136

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by bradleyoesch** [comment]
Please leave any comments to clarify/change/update/etc!

@patrick91 was there more specifically you'd like to call out other than execution order? Can/should we make this tradeoff more explicit?

File: RELEASE.md
Comment (DoctorJohn) [inline RELEASE.md:1]: We don't need a release file for changes to the docs, they're published directly after a PR is merged and don't require a full package release

File: docs/general/mutations.md
Comment (DoctorJohn) [inline docs/general/mutations.md:140]: nitpick: This PR adds a handful of links to the Apollo docs. We might want to make clear that these are thirdparty docs we're linking too to set the right expectations. Maybe by doing something like the following?

```suggestion
To avoid a graph becoming too large and to improve discoverability, it can be helpful to group mutations in a namespace, as described by [Apollos guide on Namespacing by separation of concerns](https://www.apollographql.com/docs/technotes/TN0012-namespacing-by-separation-of-concern/).
```

**Review by DoctorJohn — CHANGES_REQUESTED** [review]
Wonderfully written @bradleyoesch, thanks so much! I left two comments. Feel free to ignore the one on adjusting the linkt texts. This PR can be merged after the `RELEASE.md` file is gone.

File: docs/general/mutations.md
Comment (bradleyoesch) [reply docs/general/mutations.md]: Yep totally makes sense. Updated!

The graphql rules site is down for me right now so I couldn't find a "name" to ascribe the attribution to so I went with `[this community-created guide on GQL best practices](https://graphql-rules.com/rules/mutation-payload)`

**Review by DoctorJohn — APPROVED** [review]

**Comment by patrick91** [comment]
Thanks @DoctorJohn for reviewing this! And @bradleyoesch for the content!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**
